### ISSUE TITLE
RequireThis false positive inside annotation definition

### ISSUE THREAD

**Issue by Machine-Maker** [checkstyle/checkstyle#17464]
```bash
/var/tmp $ javac SomeAnnotation.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat checkstyle.xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="RequireThis">
      <property name="validateOnlyOverlapping" value="false"/>
    </module>
  </module>
</module>

/var/tmp $ cat SomeAnnotation.java
public @interface SomeAnnotation {

  Object A_CONSTANT_VALUE = new Object();

  String value();

  final class InnerClass {

    public void method() {
      System.out.println(A_CONSTANT_VALUE);
    }
  }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.21.1-all.jar -c checkstyle.xml SomeAnnotation.java
Starting audit...
[ERROR] /Users/jake/Projects/Java/checkstyle-test/SomeAnnotation.java:10:26: Reference to instance variable 'A_CONSTANT_VALUE' needs "SomeAnnotation.this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors
```

---

`A_CONSTANT_VALUE` isn't an instance variable, its a static one.

**Comment by Atharv3221** [comment]
I am working on this

**Comment by romani** [comment]
Fix is merged
